### PR TITLE
Moving routes to postgres

### PR DESCRIPTION
**Pull request by klepper24**
Exporting data from Mongo DB to Postgres with sql alchemy and its bulk insert.

Please pay special attention to the use of the generator, as I am not sure if this is the optimal way.

### CODE REVIEW

File: orchestrator/dags/mongo_to_postgres_routes.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+def map_routes_to_postgres_object(routes: object) -> object:
+    id_nr = 1
+    for doc in routes:
+        route = Routes(id=id_nr, line_name=doc["line_nr"] + doc["routes"]["route_nr"],
+                       line_nr=doc["line_nr"], name=doc["routes"]["route_nr"],
+                       stops_cnt=doc["routes"]["number_of_stops"],
+                       created_at=datetime.now())
+        id_nr += 1
+        yield route
```
Comment (mgierm) [inline orchestrator/dags/mongo_to_postgres_routes.py]:
```suggestion
def map_routes_to_postgres_object(routes: object) -> object:
    for id_nr, doc in enumerate(routes):
        route = Routes(id=id_nr, line_name=doc["line_nr"] + doc["routes"]["route_nr"],
                       line_nr=doc["line_nr"], name=doc["routes"]["route_nr"],
                       stops_cnt=doc["routes"]["number_of_stops"],
                       created_at=datetime.now())
        yield route
```

File: orchestrator/dags/mongo_to_postgres_routes.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            # do not commit this row
+            session.rollback()
+            for route in mapped_routes_backup:
+                session.add(route)
+                try:
+                    session.commit()
+                except IntegrityError:
+                    print("2nd try: An IntegrityError has occurred")
+                    # do not commit this row
+                    session.rollback()
```
Comment (mgierm) [inline orchestrator/dags/mongo_to_postgres_routes.py]:
```suggestion
def send_routes_to_postgres() -> None:
    routes_collection = get_mongo_collection('Routes')
    pipeline = [
        {"$unset": ["_id", "number_of_routes", "routes.stops"]},
        {"$unwind": "$routes"}
    ]
    agg_routes = routes_collection.aggregate(pipeline)

    postgres_session = create_postgres_session(settings.POSTGRES_URL)
    with postgres_session() as session:
        session.execute(text('TRUNCATE TABLE dbo.routes CASCADE'))
        session.commit()

    limit = 100
    offset = 0

    while True:
    	mapped_routes = list(map_routes_to_postgres_object(agg_routes[offset:offset + limit]))
	    with postgres_session() as session:
	        try:
	            session.bulk_save_objects(mapped_routes)
	            session.commit()
	        except IntegrityError:
	            print("1st try: An IntegrityError has occurred")
	            # do not commit this row
	            session.rollback()
	            for route in mapped_routes:
	                session.add(route)
	                try:
	                    session.commit()
	                except IntegrityError:
	                    print("2nd try: An IntegrityError has occurred")
	                    # do not commit this row
	                    session.rollback()
	    if len(mapped_routes) < limit:
	    	break
	    offset = offset + limit
```

You could try do something like this - it prevents loading all routes to memory at once - instead it loads portions of 100 routes.

File: orchestrator/dags/mongo_to_postgres_routes.py (hunk 1 above)
Comment (klepper24) [reply orchestrator/dags/mongo_to_postgres_routes.py]: good piece of advice

File: orchestrator/dags/mongo_to_postgres_routes.py (hunk 2 above)
Comment (klepper24) [reply orchestrator/dags/mongo_to_postgres_routes.py]: please check my latest solution with offset, I had to redesign your snippet a bit

Comment (mgierm) [reply orchestrator/dags/mongo_to_postgres_routes.py]: Looks good :slightly_smiling_face: :+1:

**Review by mgierm — APPROVED** [review]
:+1:

**Pull request merged**